components/add_groups_to_channel_modal/add_groups_to_channel_modal.tsx

```diff
@@ -114,9 +132,9 @@ export default class AddGroupsToChannelModal extends React.PureComponent {
         });
     }
 
-    handleSubmit = async (e) => {
+    handleSubmit = async (e: Event | GroupValue[] | undefined) => {
         if (e) {
-            e.preventDefault();
+            (e as Event).preventDefault();
```

Comment:
Why is this casting needed if we provide the event type?

Reply:
it complains with this without the casting..

Property 'preventDefault' does not exist on type 'GroupValue[]'.

Reply:
that's because `e` can also be `groupValue[]` and `undefined`, you checked for `undefined` in the previous line, but never checked against `groupValue[]`. So by casting you are asuring the compiler this has to be this type. 

If it can still be the other type (and going through `multiselect` component tells me that it is not going to be an event), it might cause trouble when trying to typecast or access it's properties/methods.

Reply:
So if it is never going to be an event i guess we can remove the whole if block and ditch the Event type ?